sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java

```diff
@@ -63,23 +64,27 @@ public static synchronized void getDbToken(
             TokenExchangeServiceCallback serviceCallback) {
         AppCenterLog.debug(LOG_TAG, "Getting a resource token from App Center...");
         String url = apiUrl + GET_TOKEN_PATH_FORMAT;
-
-        ServiceCall tokenResponse =
-            httpClient.callAsync(
-                url,
-                METHOD_POST,
-                new HashMap<String, String>() { { put(APP_SECRET_HEADER, appSecret); } },
-                    new HttpClient.CallTemplate() {
-
-                        @Override
-                        public String buildRequestBody() {
-                            return buildAppCenterGetDbTokenBodyPayload(partition);
-                        }
-
-                        @Override
-                        public void onBeforeCalling(URL url, Map<String, String> headers) { }
-                    },
-                    serviceCallback);
+        TokenResult tokenResult = TokenManager.getInstance().getToken(partition);
+        if (tokenResult != null) {
+            serviceCallback.callCosmosDb(tokenResult);
```

Comment:
I'm not sure this is going to work. Here, essentially, you are only calling `callCosmosDb` method explicitly. However, before, it was a part of the `HttpClien` "framework" in which both methods of `TokenExchangeServiceCallback` would be called within the lifecycle of the HTTP call. We need to think about how to write an equivalent of that when the token is already there.

Reply:
That is the test case for :), The `callCosmosDb` is the actual callback method that when successfully get the token from http call. If you check the `TokenManager` get token method, we directly stored the tokenResult object, and that is what it should be passed here.